src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final String DEFAULT_COLOR = "white";
+    private static final int DEFAULT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        int figurType = RANDOM.nextInt(5);
+        String color = new ColorSupplier().getRandomColor();
+        switch (figurType) {
+            case 0:
+                return new Circle(color, RANDOM.nextInt(10) + 1);
+            case 1:
+                return new Square(color, RANDOM.nextInt(10) + 1);
+            case 2:
+                return new Rectangle(color, RANDOM.nextInt(10) + 1, RANDOM.nextInt(10) + 1);
+            case 3:
+                return new RightTriangle(color, RANDOM.nextInt(10) + 1,
+                        RANDOM.nextInt(10) + 1, RANDOM.nextInt(10) + 1);
+            case 4:
+                return new IsoscelesTrapezoid(color, RANDOM.nextInt(10) + 1,
+                        RANDOM.nextInt(10) + 1, RANDOM.nextInt(10) + 1);

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For instance, instead of 'RANDOM.nextInt(10) + 1', assign this expression to a variable with a meaningful name and then pass it to the constructor.